Source/NameServer/ABRecord.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef ABRecord_H
+#define ABRecord_H
+
+#include "ABDefs.hpp"
+
+namespace AddressBookNS
+{
+   enum RecordType_t { Device = 0, DeviceExt, External, Supervisor };
+
+   // data contents factored out for serialisation 17 July 2019 ADR  
+   struct RecordData_t
+   {
+       public:
+     
+       SymAddr_t Address;          // Device's full sym address.
+       union
+       {
+           SymAddr_t Supervisor;   // Full sym address of the Device's Supervisor.
```

Comment:
I didn't really follow - how is this union disambiguated?

Reply:
From the RecordType (which appears at the end for packing reasons)